pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -610,7 +610,7 @@ func (r *ReconcileZookeeperCluster) yamlConfigMap(instance *zookeeperv1beta1.Zoo
 }
 
 func (r *ReconcileZookeeperCluster) reconcileFinalizers(instance *zookeeperv1beta1.ZookeeperCluster) (err error) {
-	if instance.Spec.Persistence.VolumeReclaimPolicy != zookeeperv1beta1.VolumeReclaimPolicyDelete {
+	if instance.Spec.Storage.Persistence != nil && instance.Spec.Storage.Persistence.VolumeReclaimPolicy != zookeeperv1beta1.VolumeReclaimPolicyDelete {
```

Comment:
why the check for " instance.Spec.Storage.Persistence != nil"

Reply:
Because in case if the storage is of type Ephemeral, "instance.Spec.Storage.Persistence" will be nil and in that case, this statement will panic